implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -63,7 +63,7 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
         Key theKey = tryAsVerificationJwk(jws);
 
         if (theKey == null) {
-            reportUnresolvableKeyException(authContextInfo.getPublicKeyContent(), authContextInfo.getPublicKeyLocation());
+            throw PrincipalMessages.msg.invalidTokenKid();
```

Comment:
@sberyozkin Thanks for feedback! I updated my PR. I changed the exception here. It seems to me that if the reason why `(theKey == null)` is different than invalid kid, some other exception should have been thrown before this? 

Eg. when loading of keys was not successful because of a connection problem, the exception gets thrown [in the constructor](https://github.com/smallrye/smallrye-jwt/blob/main/implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java#L47) already. But let me know if this is wrong assumption.

Reply:
@Skyllarr Sorry for a delay - I wonder, would it be better to introduce an exception like `UnmatchedTokenKidException` and report it as a cause of `UnresolvableKeyException` only if it is the case when 1) the token has `kid` and it is JWK set or JWK and no match is found even after a refresh... Then instead of checking the message content, we just check for the cause...

Reply:
@sberyozkin Hi, I am sorry for my delay. I have introduced an `UnmatchedTokenKidException` exception that gets thrown when the JWKs are available and kid exists but the key was not found:

`if (httpsJwks != null && httpsJwks.getJsonWebKeys() != null && jws != null && jws.getKeyIdHeaderValue() != null)`

Let me know if that looks good. Thank you!

Reply:
@Skyllarr Hi, np at all, I've badly missed your update. I'll try to prioritize on getting it in asap, will take a bit more time but I'll have a look.